lib/improver/tests/test_ensemble_copula_coupling_GeneratePercentilesFromProbabilities.py

```diff
@@ -0,0 +1,502 @@
+                     [[1.0, 0.5, 1.0,
+                       0.5, 0.5, 0.3,
+                       0.2, 0.0, 0.0]],
+                     [[1.0, 0.2, 0.5,
+                       0.2, 0.0, 0.1,
+                       0.0, 0.0, 0.0]]])
+    return set_up_spot_cube(data, "air_temperature", "1")
+
+
+class Test__add_bounds_to_thresholds_and_probabilities(IrisTest):
+
+    """Test the _add_bounds_to_thresholds_and_probabilities plugin."""
+
+    def setUp(self):
+        self.current_temperature_forecast_cube = (
+            _add_forecast_reference_time_and_forecast_period(
+                set_up_temperature_cube()))
+
+    def test_basic(self):
+        """Test that the plugin returns two numpy arrays."""
+        cube = self.current_temperature_forecast_cube
+        threshold_points = cube.coord("probability_above_threshold").points
+        probabilities_for_cdf = cube.data.reshape(3, 9)
+        bounds_pairing = (-40, 50)
+        plugin = Plugin()
+        result = plugin._add_bounds_to_thresholds_and_probabilities(
+            threshold_points, probabilities_for_cdf, bounds_pairing)
+        self.assertIsInstance(result[0], np.ndarray)
+        self.assertIsInstance(result[1], np.ndarray)
+
+    def test_bounds_of_threshold_points(self):
+        """
+        Test that the plugin returns the expected results for the
+        threshold_points, where they've been padded with the values from
+        the bounds_pairing.
+        """
+        cube = self.current_temperature_forecast_cube
+        threshold_points = cube.coord("probability_above_threshold").points
+        probabilities_for_cdf = cube.data.reshape(3, 9)
+        bounds_pairing = (-40, 50)
+        plugin = Plugin()
+        result = plugin._add_bounds_to_thresholds_and_probabilities(
+            threshold_points, probabilities_for_cdf, bounds_pairing)
+        self.assertArrayAlmostEqual(result[0][0], bounds_pairing[0])
+        self.assertArrayAlmostEqual(result[0][-1], bounds_pairing[1])
+
+    def test_probability_data(self):
+        """
+        Test that the plugin returns the expected results for the
+        probabilities, where they've been padded with zeros and ones to
+        represent the extreme ends of the Cumulative Distribution Function.
+        """
+        cube = self.current_temperature_forecast_cube
+        threshold_points = cube.coord("probability_above_threshold").points
+        probabilities_for_cdf = cube.data.reshape(3, 9)
+        zero_array = np.zeros(probabilities_for_cdf[:, 0].shape)
+        one_array = np.ones(probabilities_for_cdf[:, 0].shape)
+        bounds_pairing = (-40, 50)
+        plugin = Plugin()
+        result = plugin._add_bounds_to_thresholds_and_probabilities(
+            threshold_points, probabilities_for_cdf, bounds_pairing)
+        self.assertArrayAlmostEqual(result[1][:, 0], zero_array)
+        self.assertArrayAlmostEqual(result[1][:, -1], one_array)
+
+
+class Test__probabilities_to_percentiles(IrisTest):
```

Comment:
I think that some of these tests could use less input data to make them clearer, but I'll talk to you about them.